apps/src/templates/sectionsRefresh/coteacherSettings/AddCoteacher.jsx

```diff
@@ -10,8 +10,58 @@ import Button from '@cdo/apps/templates/Button';
 
 import styles from './coteacher-settings.module.scss';
 
+export const getInputErrorMessage = (email, coteachersToAdd, sectionId) => {
+  if (email === '') {
+    return Promise.resolve(i18n.coteacherAddNoEmail());
+  }
+  if (!isEmail(email)) {
+    return Promise.resolve(i18n.coteacherAddInvalidEmail({email}));
+  }
+  if (coteachersToAdd.some(coteacher => coteacher === email)) {
+    return Promise.resolve(i18n.coteacherAddAlreadyExists({email}));
+  }
+
+  return fetch(
+    `/api/v1/section_instructors/check?email=${encodeURIComponent(email)}` +
+      (sectionId && `&section_id=${sectionId}`),
+    {
+      type: 'GET',
+      headers: {
+        'Content-Type': 'application/json; charset=utf-8',
+      },
+    }
+  ).then(response => {
+    if (response.ok) {
+      return '';
+    }
+    if (response.errorThrown === 'Not Found') {
+      return i18n.coteacherAddNoAccount({email});
+    }
+    if (response.errorThrown === 'Forbidden') {
+      return i18n.coteacherUnableToEditCoteachers();
+    }
+
+    return response.json().then(json => {
+      if (json.error.includes('already invited')) {
+        return i18n.coteacherAddAlreadyExists({email});
+      }
+      if (json.error.includes('section full')) {
```

Comment:
Would it be good to also do some client-side validation on the number of co-teachers so that this message can be displayed even for new sections, or does the other logic to disable the buttons at the limit cover that case well enough?

Reply:
I think the button being disabled is good enough, there shouldn't actually be a way to see this message, but I wanted to add it just in case.